### PR TITLE
Make the search bar auto complete

### PR DESCRIPTION
**Pull request by jadmnsaad**
https://app.zenhub.com/workspaces/epictrack-63891ea941d309001fa292cf/issues/gh/bcgov/epic.track/1282

Details:
- Use an auto complete component
- Load options with list type projects with works
- Add a find all for projects with works
- Add the works field for project, lazy loaded
- Add with_works parameter for poject get all endpoint

### CODE REVIEW

File: epictrack-web/src/components/myWorkplans/Filters/NameFilter.tsx [hunk 1]
```diff
@@ -20,25 +28,64 @@ export const NameFilter = () => {
...
   };
 
-  const debouncedSetSearchOptions = useRef(
-    debounce((searchText: string) => {
-      handleSearchOptions(searchText);
-    }, 1000)
-  ).current;
+  // Fetch project names from the backend when searchText changes
+  useEffect(() => {
+    const fetchProjectNames = async () => {
```
Comment (TomChapmanGov) [inline epictrack-web/src/components/myWorkplans/Filters/NameFilter.tsx:33]: Could we declare this outside of useffect?

File: epictrack-web/src/services/projectService/projectService.ts [hunk 2]
```diff
@@ -5,8 +5,11 @@ import { MasterBase } from "../../models/type";
 import { ListType } from "../../models/code";
 
 class ProjectService implements ServiceBase {
-  async getAll(return_type?: string) {
-    return await http.GetRequest(Endpoints.Projects.PROJECTS, { return_type });
+  async getAll(return_type?: string, with_works?: boolean) {
```
Comment (saravanpa-aot) [inline epictrack-web/src/services/projectService/projectService.ts:8]: with_works? is includeWorks better?

File: epictrack-web/src/components/myWorkplans/Filters/NameFilter.tsx [hunk 3]
```diff
@@ -20,25 +28,64 @@ export const NameFilter = () => {
...
+      // Replace this with your actual backend API call
```
Comment (TomChapmanGov) [inline epictrack-web/src/components/myWorkplans/Filters/NameFilter.tsx:34]: Is this comment still relevant?

File: epictrack-web/src/services/projectService/projectService.ts (hunk 2 above)
Comment (saravanpa-aot) [reply epictrack-web/src/services/projectService/projectService.ts:8]: i am not sure you want project which has a work or you wanna include a work ..so plz feel to disregard this

Comment (saravanpa-aot) [reply epictrack-web/src/services/projectService/projectService.ts:8]: return_type ? is it the project type?

Comment (jadmnsaad) [reply epictrack-web/src/services/projectService/projectService.ts:8]: I need only the project with works, I don't need to include the works info.

Comment (jadmnsaad) [reply epictrack-web/src/services/projectService/projectService.ts:8]: return_type  is not my addition, it's already there. It decides what schema to use

File: epictrack-web/src/components/myWorkplans/Filters/NameFilter.tsx (hunk 3 above)
Comment (jadmnsaad) [reply epictrack-web/src/components/myWorkplans/Filters/NameFilter.tsx:34]: not, I should remove this

File: epictrack-web/src/components/myWorkplans/Filters/NameFilter.tsx (hunk 1 above)
Comment (jadmnsaad) [reply epictrack-web/src/components/myWorkplans/Filters/NameFilter.tsx:33]: I do that usually, but this is actually the better pattern when dealing with an asynchronous effect. From chatgpt:
image

Comment (TomChapmanGov) [reply epictrack-web/src/components/myWorkplans/Filters/NameFilter.tsx:33]: Interesting I didnt know that

**Review by TomChapmanGov — APPROVED** [review]

**Pull request merged**